dulwich/objects.py

```diff
@@ -1510,6 +1510,18 @@ def sign(self, keyid: Optional[str] = None) -> None:
                     self.as_raw_string(), mode=gpg.constants.sig.mode.DETACH
                 )
 
+    def without_sig(self) -> "Commit":
+        """Return a Commit object copy with the signature (GPG or SSH) removed.
+
+        self.gpgsig is a signature for the raw string serialization bytes of the
+        Commit object returned.
+        """
+        ret = self.copy()
+        assert isinstance(ret, Commit)
+        ret._gpgsig = None
+        ret.gpgsig = None
```

Comment:
Are you sure you want that removed? I'm repeating the code that has been on the master master branch prior to this PR.

Reply:
in that case it's fine because the object isn't returned and discarded after verifying the signature. here, the caller might rely on Commit.gpgsig actually working later.